src/style-compiler/index.js

```diff
@@ -0,0 +1,58 @@
+const postcss = require('postcss')
+const defaults = require('lodash.defaultsdeep')
+
+const trim = require('./plugins/trim')
+const scopeId = require('./plugins/scope-id')
+
+function compileStyle (style, filename, config) {
+  const plugins = [trim].concat(config.plugins)
+  const options = Object.assign({
+    to: filename,
+    from: filename
+  }, config.options)
+
+  // source map
+  if (config.needMap && !style.map) {
+    if (!style.map) {
+      throw Error('Previous source map is missing.')
+    }
+
+    options.map = {
+      inline: false,
+      annotation: false,
+      prev: style.map
+    }
+  }
+
+  if (!style.descriptor) {
+    throw Error('SFC block descriptor is missing.')
+  }
+
+  // add plugin for scoped css rewrite
+  if (style.descriptor.scoped) {
+    if (typeof (config.scopeId) !== 'string') {
+      throw Error(`'scopeId' is required to compile scoped style.`)
+    }
+    plugins.push(scopeId({ id: config.scopeId }))
+  }
+
+  return postcss(plugins).process(style.code, options).then(result => {
```

Comment:
Calling `process.css` will give us css, and `process.map` gives you the map. The result of process is a [LazyResult](http://api.postcss.org/LazyResult.html), which is synchronous

Reply:
Okay. I'll make changes.